### PR TITLE
remote: allow local options for repo remotes

### LINKED ISSUE #4276: remote local config does not see existing remote
### Please provide information about your setup

**Output of `dvc version`:**

```console
DVC version: 1.1.10+1c9588
Python version: 3.8.4
Platform: macOS-10.15.4-x86_64-i386-64bit
Binary: False
Package: None
Supported remotes: azure, gdrive, gs, hdfs, http, https, s3, ssh, oss
Repo: dvc, git
Filesystem type (workspace): ('apfs', '/dev/disk1s1')
```

**Additional Information (if any):**

```
dvc remote add -d storage s3://storage
dvc remote modify storage region us-east-2 --local
```

complains:

```
ERROR: configuration error - config file error: remote 'storage' doesn't exists.
```

For example, it breaks the use case of using `dvc remote modify` to store credentials in the local config only.

### PR DESCRIPTION
**Pull request by efiop**
Fixes #4276

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/func/test_remote.py [hunk 1]
```diff
@@ -232,6 +232,13 @@ def test_modify_missing_remote(tmp_dir, dvc):
     assert main(["remote", "modify", "myremote", "user", "xxx"]) == 251
 
 
+def test_modify_parent_remote(tmp_dir, dvc):
```
Comment (skshetry) [inline tests/func/test_remote.py:235]: What's a parent remote? Just find it a bit confusing.

File: tests/func/test_remote.py [hunk 2]
```diff
@@ -232,6 +232,13 @@ def test_modify_missing_remote(tmp_dir, dvc):
...
+    assert main(["remote", "add", "myremote", "http://example.com/path"]) == 0
+    assert (
+        main(["remote", "modify", "myremote", "user", "xxx", "--local"]) == 0
+    )
```
Comment (skshetry) [inline tests/func/test_remote.py:239]: Should we also try reading the config? I understand that the issue was during `modify` but I don't want to trust that.

File: tests/func/test_remote.py (hunk 1 above)
Comment (efiop) [reply tests/func/test_remote.py:235]: Yeah, called it parent for a lack of better word. In this case it means that .dvc/config is a parent config for .dvc/config.local

Comment (skshetry) [reply tests/func/test_remote.py:235]:
```suggestion
def test_remote_modify_local_on_repo_config(tmp_dir, dvc):
```

File: tests/func/test_remote.py (hunk 2 above)
Comment (efiop) [reply tests/func/test_remote.py:239]: Good point! Added more throughout checks.

**Pull request merged**